### PR TITLE
Footer preview!

### PR DESCRIPTION
**Pull request by wiredearp**
@zdlm @sampi

There's a functional Footer in here, but the Docs are not quite up for it yet and there's still a few (or a lot) design tweaks that need to go in, however @lloydhumphreys would like to be able to show a preview to some of the guys down in Bucharest, so here it is. The Footer features:

- buttons as usual
- actions for selections
- a status message
- a strange checkbox
- some kind of Pager
- one Configuration button
- one Collaboration button

My first attempt at all this proved to be quite complicated (code-wise) because I would build the Footer using spirits instead of models. Since the whole setup will need to be duplicated into the Header (in a future project), I then refactored everything to use models instead of spirits and uncovered some bugs in the framework (or rather, in the imaginary specification of the framework) so there is quite a lot of code that has changed. Sorry about that.

I will attempt to `rebase` all this at some point, it however didn't pan out so well now that `master` had been subjected to an alternative version of Prettier and so I got caught in some kind of endless rebase loop :confused:

### CODE REVIEW

File: docs/src/xhtml/components/bars/statusbar.xhtml [hunk 1]
```diff
@@ -3,16 +3,25 @@
...
+		}	
+		</script>
 	</head>
 	<body>
 		<header data-ts="TopBar"></header>
 		<main data-ts="Main">
 			<div data-ts="Panel" data-ts.label="Main StatusBar" data-ts.onselect="ts.ui.StatusBar.show()">
 				<article>
-					<h1>Main StatusBar</h1>
+					<h1>Main StatusBar is deprecated 😞</h1>
```
Comment (zdlm) [inline docs/src/xhtml/components/bars/statusbar.xhtml]: :rofl:

File: src/runtime/js/ts.ui/bars/bars-gui@tradeshift.com/api/ts.ui.Footer.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		 */
+		checkbox: chained(function(json) {
+			if (arguments.length) {
+				bar().checkbox(json);
+			} else {
+				return bar().checkbox();
+			}
+		}),
+
+		/**
```
Comment (zdlm) [inline src/runtime/js/ts.ui/bars/bars-gui@tradeshift.com/api/ts.ui.Footer.js]: Some comments

**Comment by wiredearp** [comment]
:warning: @lloydhumphreys Our automated test infrastructure appears to have suffered a total collapse and so we won't be able to release anything at all today :cry:

**Comment by wiredearp** [comment]
:goat: safe :/

**Pull request merged**